airbyte-webapp/src/components/connectorBuilder/manifestToBuilderForm.ts

```diff
@@ -316,9 +323,22 @@ function manifestAuthenticatorToBuilder(
     builderAuthenticator = {
       type: "NoAuth",
     };
+  } else if (manifestAuthenticator.type === undefined) {
+    throw new ManifestCompatibilityError(streamName, "authenticator has no type");
   } else if (manifestAuthenticator.type === "CustomAuthenticator") {
     throw new ManifestCompatibilityError(streamName, "uses a CustomAuthenticator");
   } else if (manifestAuthenticator.type === "OAuthAuthenticator") {
+    if (
+      Object.values(manifestAuthenticator.refresh_request_body ?? {}).filter(
+        (val) => (val !== null && typeof val === "object") || Array.isArray(val)
```

Comment:
Ultra nit-picky - the only values we support are strings but this logic will let numbers and booleans through. The behavior is the following:
* As long as you don't touch the input, the type is preserved
* As soon as you type in the value field, it will implicitly cast to string

Seems cleaner to just throw on this until we actually support it.

Reply:
That makes sense, I agree we should just throw on non-string values for now. Updated